nuget/lib/dependabot/nuget/file_parser/project_file_parser.rb

```diff
@@ -131,27 +166,20 @@ def add_transitive_dependencies(project_file, doc, dependency_set)
         end
 
         def add_transitive_dependencies_from_project_references(project_file, doc, dependency_set)
-          project_file_directory = File.dirname(project_file.name)
-          is_rooted = project_file_directory.start_with?("/")
-          # Root the directory path to avoid expand_path prepending the working directory
-          project_file_directory = "/" + project_file_directory unless is_rooted
-
           # Look for regular project references
-          doc.css(PROJECT_REFERENCE_SELECTOR).each do |reference_node|
+          project_refs = doc.css(PROJECT_REFERENCE_SELECTOR)
+          # Look for ProjectFile references (dirs.proj)
+          project_files = doc.css(PROJECT_FILE_SELECTOR)
+          ref_nodes = project_refs + project_files
+
+          ref_nodes.each do |reference_node|
             relative_path = dependency_name(reference_node, project_file)
             # This could result from a <ProjectReference Remove="..." /> item.
             next unless relative_path
 
-            # normalize path separators
-            relative_path = relative_path.tr("\\", "/")
-            # path is relative to the project file directory
-            relative_path = File.join(project_file_directory, relative_path)
-
-            # get absolute path
-            full_path = File.expand_path(relative_path)
-            full_path = full_path[1..-1] unless is_rooted
+            full_project_path = full_path(project_file, relative_path)
 
-            referenced_file = dependency_files.find { |f| f.name == full_path }
+            referenced_file = dependency_files.find { |f| f.name == full_project_path }
```

Comment:
@brettfo Has this been subtly wrong for a while? the `dependency_file.name` will be relative to the `dependency_file.directory` not an absolute path, right?

Reply:
If that were the case wouldn't this test fail? https://github.com/dependabot/dependabot-core/blob/ec0349e8c0793612b46e60021f816709759b71ff/nuget/spec/dependabot/nuget/file_parser/project_file_parser_spec.rb#L169

I did some local testing and it seem like "full_path" in this case is actually working relative to the repo-root instead of the drive root, which is the same thing that the DepsFiles are doing so this works.

Reply:
ah ok. It was `expand_path` which was causing the question. Reading the docs answers it.